### DISCUSSION TITLE
Convert Inline Footnotes to Regular Footnotes in Markdown | /en/2023/02/markdown-footnotes/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1473] [Comments]
After using Hugo 0.25.1 for my personal website for nearly six years, I have finally decided to upgrade it. I&rsquo;ll explain the reasons later. One problem I encountered was that I had to convert &hellip;

https://yihui.org/en/2023/02/markdown-footnotes/

**Comment by chuxinyuan** [comment]
建议 Posit 定制一个 Hugo，安安静静的工作就好，没必要经常更新。

**Reply by yihui** [reply]
你不需要更新啊。用 `blogdown::install_hugo()` 安装了一个能用的版本之后，就可以用它到地老天荒。我在这篇日志开头说了我用一个极老的 Hugo 版本撑了六年毫无压力；结尾也说了我并没有很强的理由需要更新，普通用户更没有必要更新。

**Reply by chuxinyuan** [reply]
一来国内的环境下载 Hugo 有点麻烦，二来默认的安装的 Hugo 是最新版本的，越来越臃肿了，所以就想能否在安装 blogdown 包的时候自动安装一个定制版的轻量的 Hugo。

**Reply by yihui** [reply]
这个很难选择“一个”，因为备选有一百多个，我很难替用户选出一个来。你说的是对的，新版本确实越来越臃肿。

国内下载 Github 文件麻烦，这个问题我也没办法……

**Comment by chuxinyuan** [comment]
一点小小的建议：
1. 超链接可以考虑换成浅灰色的，满屏的虚线影响视觉效果，鼠标放在上面变成黑色的实线，相当于反馈给读者一个信息：“你已选中了超链接，可以点了”。
2. 评论的颜色是否考虑从 dark 改为 light，和你的博客的画风保持一致。
3. 鼠标放在代码块上时代码块右上角显示一个复制的图标，点击后自动复制代码到剪切板。

**Reply by yihui** [reply]
2. 评论区的颜色根据你的系统主题偏好来设定的：https://github.com/yihui/yihui.org/blob/756e0e671630452c97e15db5c40db680f5f6c0df/layouts/partials/comments.html#L14

**Reply by chuxinyuan** [reply]
嗯，手机上试了下确实是跟随系统的。但是有个问题，你的博客并不是跟随读者系统主题的啊，所以我觉得评论是你博客的一部分，跟随你的博客主题就好，一体感更强点。

**Reply by yihui** [reply]
理论上当然我的博客本身最好也自适应读者系统主题，不过我没时间折腾这件事。

**Reply by yihui** [reply]
4c1f99f 已经开启自动暗黑模式（跟随系统主题）。

**Reply by chuxinyuan** [reply]
> 已经开启自动暗黑模式（跟随系统主题）。

原来连“房子“也换了啊

**Comment by chuxinyuan** [comment]
我发现这个主题的段落没有两端对齐，最后通过如下代码搞定了：
```css
p {
  text-align: justify;
  text-justify: inter-word;
}
```

**Reply by yihui** [reply]
在浏览器中，两端对齐对中英文混排的文章来说容易出麻烦，就是当一行的字数不太多也不太少时，两端对齐可能会硬性拉扯字符到两端，导致字符间距特别大，很难看。

**Reply by chuxinyuan** [reply]
是的，确实有你说的这种情况。我打算牺牲“少数人”，成全“大多数人”。

**Reply by chuxinyuan** [reply]
> 当一行的字数不太多也不太少时，两端对齐可能会硬性拉扯字符到两端，导致字符间距特别大

这种情况在我的手机上（华为浏览器）比较明显，在 Edge 浏览器上还好。
